presto-main/src/main/java/com/facebook/presto/split/BufferingSplitSource.java

```diff
@@ -45,6 +45,12 @@ public ConnectorId getConnectorId()
     }
 
     @Override
+    public Split getEmptySplit()
```

Comment:
Could elaborate? I think I can not create one instance of `emptySplitSource` per query, because I need both `connectorId` and `transactionHandle` to instantiate an `EmptySplit` and they must (should?) match those created by real split source. 

I have added `getTransactionHandle()` method to the `SplitSource` interface (along already existing `getConnectorId()` and I have inlined creation of `EmptySplit` to `SourceParitionedScheduler`.